### PR TITLE
Adding accessibility section to the guides

### PR DESCRIPTION
**Pull request by MelSumner**
If merged, this will add an accessibility section to the guides, and will resolve issue #368.

### CODE REVIEW

File: guides/v3.7.0/reference/accessibility.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+The purpose of this guide is to give every Ember developer a starting point for making your Ember application minimally accessible. As accessibility can be a complex issue, it should be noted that this guide only covers a some of the basics, and should not be considered a comprehensive guide.
```
Comment (jenweber) [inline guides/v3.7.0/reference/accessibility.md]: This intro would benefit from a link about web accessibility in general.

File: guides/v3.7.0/reference/accessibility.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
+The purpose of this guide is to give every Ember developer a starting point for making your Ember application minimally accessible. As accessibility can be a complex issue, it should be noted that this guide only covers a some of the basics, and should not be considered a comprehensive guide. 
+
+## Screen Reader and Browser Combinations
+
+There are assistive technologies (known as screen readers) available for all common desktop platforms and mobile devices.
```
Comment (jenweber) [inline guides/v3.7.0/reference/accessibility.md]:
```suggestion
One important aspect of accessibility is whether someone using assistive technology can navigate and interact with your app. There are assistive technologies (known as screen readers) available for all common desktop platforms and mobile devices. 
```

File: guides/v3.7.0/reference/accessibility.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+- IE & JAWS (Windows)
+- Safari & VoiceOver (MacOS)
+
+The absolute best method for learning how a screen reader works is using one yourself! It might feel a little awkward at first, but understanding how to use a screen reader (and other assistive technology) will help you become a more skilled developer. 
+
+## Ember Application Configuration
+
+### Ember Application vs role="application"
+
+The first thing that should be discussed in this section is this: "application" in Ember development and "application" in landmark roles have two _very_ different meanings.
```
Comment (jenweber) [inline guides/v3.7.0/reference/accessibility.md]:
```suggestion
It is important to note that "application" in Ember development and "application" in landmark roles have two _very_ different meanings. 
```

File: guides/v3.7.0/reference/accessibility.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+The absolute best method for learning how a screen reader works is using one yourself! It might feel a little awkward at first, but understanding how to use a screen reader (and other assistive technology) will help you become a more skilled developer. 
+
+## Ember Application Configuration
+
+### Ember Application vs role="application"
+
+The first thing that should be discussed in this section is this: "application" in Ember development and "application" in landmark roles have two _very_ different meanings. 
+
+The TL;DR? Don't use `role="application"` until you have done your research and know exactly how it is to be used correctly (if at all). There are very few use cases where the role of application is appropriate.
```
Comment (jenweber) [inline guides/v3.7.0/reference/accessibility.md]:
```suggestion
To summarize, don't use `role="application"` until you have done your research and know exactly how it is to be used correctly (if at all). There are very few use cases where the role of application is appropriate. 
```

File: guides/v3.7.0/reference/accessibility.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+The first thing that should be discussed in this section is this: "application" in Ember development and "application" in landmark roles have two _very_ different meanings. 
+
+The TL;DR? Don't use `role="application"` until you have done your research and know exactly how it is to be used correctly (if at all). There are very few use cases where the role of application is appropriate. 
+
+Read more about it: [https://a11yproject.com/posts/how-to-use-application-role/](https://a11yproject.com/posts/how-to-use-application-role/)
+
+
+### Optional Feature: Application Template Wrapper
+
+If you are using the [application template wrapper](#toc_optional-features) enabled (default state), then you will need to add certain aria roles to your [landmark regions](https://www.w3.org/WAI/PF/aria/roles#landmark_roles), even if you are using native HTML elements, because those regions are not the direct child descendant of the body element (they are the children of the div that wraps the Ember app).
```
Comment (jenweber) [inline guides/v3.7.0/reference/accessibility.md]: It's not clear from this description whether having the application template wrapper is a good thing or not. I think it should recommend for or against it as a way to improve accessibility, right away. In general, I would say always show first the option that is the "better" choice.

File: guides/v3.7.0/reference/accessibility.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  <main></main>
+  <footer></footer>
+</body>
+```
+
+> Note: To learn more about landmark roles and how to use them: [https://www.w3.org/WAI/PF/aria/roles#landmark_roles](https://www.w3.org/WAI/PF/aria/roles#landmark_roles). Still need more help? Visit the #topic-a11y channel in [Ember chat](https://emberjs.com/community/). 
+
+## Basic Guidelines for Accessibility
+
+This section will become more detailed over time, but to start, here is a checklist of some things to keep in mind when developing your application:
```
Comment (jenweber) [inline guides/v3.7.0/reference/accessibility.md]:
```suggestion
Here is a checklist of some things to keep in mind when developing your application:
```

File: guides/v3.7.0/reference/accessibility.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+- In general, avoid making your own keyboard shortcuts. [Screen readers already provide quite a few.](https://dequeuniversity.com/screenreaders/) There is some nuance here, so proceed with caution should you choose to do so. 
+- "Completely accessible" may be somewhat of a misnomer. Practical accessibility looks more like 90% coding to the spec and 10% filing browser bugs (or keeping track of existing browser bugs). Keep in mind that if you choose to implement a workaround for a browser bug, you will need to to put an issue in your product backlog to follow up on browser bugs at a later date. 
+
+### Focus
+Focus management is a large part of how your application's code coordinates with the code that runs screen readers (If you peek through the source code for [NVDA](https://github.com/nvaccess/nvda), you'll see what I mean).
+
+Again, we intend for this section to have more information but to get you started, here are some focus basics:
```
Comment (jenweber) [inline guides/v3.7.0/reference/accessibility.md]:
```suggestion
To get you started, here are some focus basics: 
```

File: guides/v3.7.0/reference/accessibility.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+  - All interactive elements must have an accessible name.
+- The values for the role attribute are pre-defined by the ARIA specification. This is not something an author can define a custom value for (that is not listed in the spec). [Learn more about roles in the specification.](https://www.w3.org/TR/wai-aria/#roles_categorization)
+- In general, avoid making your own keyboard shortcuts. [Screen readers already provide quite a few.](https://dequeuniversity.com/screenreaders/) There is some nuance here, so proceed with caution should you choose to do so. 
+- "Completely accessible" may be somewhat of a misnomer. Practical accessibility looks more like 90% coding to the spec and 10% filing browser bugs (or keeping track of existing browser bugs). Keep in mind that if you choose to implement a workaround for a browser bug, you will need to to put an issue in your product backlog to follow up on browser bugs at a later date. 
+
+### Focus
+Focus management is a large part of how your application's code coordinates with the code that runs screen readers (If you peek through the source code for [NVDA](https://github.com/nvaccess/nvda), you'll see what I mean).
```
Comment (jenweber) [inline guides/v3.7.0/reference/accessibility.md]: Could use a link or a super brief description of what focus is

File: guides/v3.7.0/reference/accessibility.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+Again, we intend for this section to have more information but to get you started, here are some focus basics: 
+- There's a difference between browse mode and focus mode in screen readers- see ["Focus Please"](https://codepen.io/melsumner/live/ZJeYoP).
+- While we have an [RFC](https://github.com/emberjs/rfcs/pull/433) to address accessible routing in Ember more permanently, in the meantime there are a few addons to help address the primary focus flaw that affects screen reader users. Evaluate [ember-a11y](https://github.com/ember-a11y/ember-a11y) and [ember-self-focused](https://github.com/linkedin/self-focused/tree/master/packages/ember-self-focused) to see which approach might fit your application best. 
+- If you have an element in your application that opens a modal (like, say, a button element), focus should return to that same element once that modal is closed. 
+
+## Ember Addon Support
+
+There are some existing Ember addons that may help you make your app more accessible. Each addon should be evaluated for its own usefulness and merit- you may find in some instances, that it would be better to implement the ideas presented in the addon in your own application.
```
Comment (jenweber) [inline guides/v3.7.0/reference/accessibility.md]: I think this should have a "community maintained" disclaimer and link to an "accessibility" search on Ember Observer "to find more addons."

File: guides/v3.7.0/reference/accessibility.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
+The purpose of this guide is to give every Ember developer a starting point for making your Ember application minimally accessible. As accessibility can be a complex issue, it should be noted that this guide only covers a some of the basics, and should not be considered a comprehensive guide. 
+
+## Screen Reader and Browser Combinations
```
Comment (jenweber) [inline guides/v3.7.0/reference/accessibility.md]: I am wondering whether this should be the first section or not. If someone only reads this far, have they received the most important info, or would you rather other parts take precedence?

File: guides/release/reference/accessibility.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+## Screen Reader and Browser Combinations
+
+There are assistive technologies (known as screen readers) available for all common desktop platforms and mobile devices. 
+
+- VoiceOver, integrated in Apple products
+- Narrator, integrated in Windows products
+- Orca, available for integration in Ubuntu, otherwise available as a download
+- JAWS, proprietary software by Freedom Scientific, available for Windows
+- NVDA, open source software, available for Windows
+- Talkback, integrated in Android products
```
Comment (drewlee) [inline guides/release/reference/accessibility.md]: Just a minor nitpick, "Talkback" should be denoted as "TalkBack".

**Pull request closed without merging**